File: 2058/2058.cs
public class Solution {
    public int FindTheMinimumAndMaximumNumber(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(FindTheMinimumAndMaximumNumber(root.left), FindTheMinimumAndMaximumNumber(root.right));
    }
}

File: 2058/2058.go
func findTheMinimumAndMaximumNumber(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := findTheMinimumAndMaximumNumber(root.Left)
	r := findTheMinimumAndMaximumNumber(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}